src/DeductionManager.cpp

```
@@ -0,0 +1,164 @@
+    // Use measured weight if available, otherwise target weight
+    float remaining = (ot3d.measured_filament_weight_g > 0)
+                      ? ot3d.measured_filament_weight_g
+                      : (float)ot3d.target_weight_g;
+
+    float deduction = (pending > remaining) ? remaining : pending;
+
+    // Subtract from the weight field the tag uses
+    if (ot3d.measured_filament_weight_g > 0) {
+        ot3d.measured_filament_weight_g -= (uint16_t)deduction;
+    } else {
+        int newWeight = (int)ot3d.target_weight_g - (int)deduction;
```

Clang (14.0.6)

[warning] variable 'newWeight' is not initialized

(cppcoreguidelines-init-variables)